### PR TITLE
refactoring as per conversation with Simon

### PR DESCRIPTION
**Pull request by somanole**
@kassapo @simonredfern 
please consider the changes.

Refactoring includes:
- get rid of private/public type methods. 
- versioning 
- introduction on the Context
- fix tests

### CODE REVIEW

File: obp-ri-transport/src/main/java/com/tesobe/obp/transport/spi/ConnectorV0.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    log.trace("{} {}", request, response);
+
+    return decoder.accounts(response);
+  }
+
+  @Override public Optional<Bank> getBank(String bankId, Context context)
+    throws InterruptedException, DecoderException
+  {
+    String request = encoder.getBank(context, bankId).toString();
+    String response = sender.send(new Message("id", request));
```
Comment (kassapo) [inline obp-ri-transport/src/main/java/com/tesobe/obp/transport/spi/ConnectorV0.java]: "id" is bad - see methods below

File: obp-ri-transport/src/main/java/com/tesobe/obp/transport/spi/ConnectorV0.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    return decoder.account(response);
+  }
+
+  @Override
+  public Iterable<Account> getAccounts(String bankId, Context context)
+    throws InterruptedException, DecoderException
+  {
+    String request = encoder.getAccounts(context, bankId).toString();
+    String response = sender.send(new Message("id", request));
```
Comment (kassapo) [inline obp-ri-transport/src/main/java/com/tesobe/obp/transport/spi/ConnectorV0.java]: "id" is bad - see methods below

File: obp-ri-transport/src/main/java/com/tesobe/obp/transport/spi/ConnectorV0.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
+// Copyright
```
Comment (kassapo) [inline obp-ri-transport/src/main/java/com/tesobe/obp/transport/spi/ConnectorV0.java]: Wrong Copyright

File: obp-ri-transport/src/test/java/com/tesobe/obp/transport/spi/ConnectorV0Test.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      }
+    });
+  }
+
+  @After public void shutdown()
+  {
+    service.shutdown();
+  }
+
+  @Test public void getAccount() throws Exception
```
Comment (kassapo) [inline obp-ri-transport/src/test/java/com/tesobe/obp/transport/spi/ConnectorV0Test.java:69]: You lost many tests - you should have at least two tests for each getXXX : with and without valid user

**Comment by simonredfern** [comment]
instead of Context, I wonder if we should have OutboundContext (because we might have InboundContext coming from the south side)?

Then 
OutboundContext comes from OBP
InboundContext comes from South side

what do you think?

**Comment by somanole** [comment]
I think that's a good idea, will check-in the changes in a bit.

**Pull request closed without merging**